### PR TITLE
Implement matmul and gemm

### PR DESCRIPTION
**Pull request by huningxin**

### CODE REVIEW

**Comment by huningxin** [comment]
@BruceDai , PTAL. Thanks.

File: src/reshape.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+import {Tensor, sizeOfShape} from './tensor.js';
+
+/**
+ *
```
Comment (BruceDai) [inline src/reshape.js:6]: @huningxin Please add the function's description,  likes 'Alter the shape of a tensor to a given new shape. '

**Review by BruceDai — COMMENTED** [review]
LGTM with a nit.

File: src/reshape.js (hunk 1 above)
Comment (huningxin) [reply src/reshape.js]: good catch, fixed in the new commit, PTAL. Thanks.

Comment (BruceDai) [reply src/reshape.js]: LGTM, thanks.

**Pull request merged**